### ISSUE TITLE
Fresh install of DRP on Ubuntu 18.04 only listens on IPv6

### ISSUE THREAD

**Issue by tombull** [digitalrebar/provision#1240]
I installed DRP on a fresh install of Ubuntu 18.04.2 using the command:
`curl -fsSL get.rebar.digital/stable | bash -s -- install`
I then started the service using the commands:
```
sudo systemctl daemon-reload && sudo systemctl enable dr-provision
sudo systemctl daemon-reload && sudo systemctl start dr-provision
```

I couldn't connect to the machine on https://machine:8092/, so I then ran:
`netstat -tulpn | grep LISTEN`
The only lines referring to dr-provision are:
```
tcp6       0      0 :::8080                 :::*                    LISTEN      3101/dr-provision   
tcp6       0      0 :::8091                 :::*                    LISTEN      3101/dr-provision   
tcp6       0      0 :::8092                 :::*                    LISTEN      3101/dr-provision
```
There are no listens on tcp - only tcp6. The dr-provision service is not listening on IPv4, only IPv6. What can I do to get the dr-provision service to listen on IPv4.

**Comment by galthaus** [comment]
Is this failing?  This is looks normal for systems with dual stacks enabled, I think.

**Comment by tombull** [comment]
I don't know if this is expected behaviour, but my naive expectation was that it would listen on IPv4 as well as IPv6 - and if it doesn't, then at least have an option to listen on IPv4.

**Comment by tombull** [comment]
If you're asking if the output is normal for a system with both IPv4 and IPv6, where the service is listening on both IPv4 and IPv6, then it's not. A service that listens on both IPv4 and IPv6 should look something like:
```
tcp        0      0 0.0.0.0:22              0.0.0.0:*               LISTEN      877/sshd            
tcp6       0      0 :::22                   :::*                    LISTEN      877/sshd 
```

**Comment by galthaus** [comment]
I have this:
[root@greg-master-1 ~]# netstat -an | grep 8092
tcp        0      0 127.0.0.1:48682         127.0.0.1:8092          ESTABLISHED
tcp        0      0 147.75.66.61:41752      147.75.65.105:8092      ESTABLISHED
tcp        0      0 147.75.66.61:41756      147.75.65.105:8092      ESTABLISHED
tcp        0      0 147.75.66.61:41750      147.75.65.105:8092      ESTABLISHED
tcp        0      0 147.75.66.61:41736      147.75.65.105:8092      ESTABLISHED
tcp6       0      0 :::8092                 :::*                    LISTEN     
tcp6       0      0 127.0.0.1:8092          127.0.0.1:48682         ESTABLISHED

And it works for me.   SSH is special and may open two explicit sockets.  Regardless, try and lets see what happens.  Can you attach to the UX through the https://<Ip:port> URL?

**Comment by tombull** [comment]
I can access the DRP service on 8092 from the machine itself using something like `curl --insecure https://127.0.0.1:8092`. I can also port forward a port from my client laptop to the machine running DRP using the command `ssh -L 8092:localhost:8092 machine`, then accessing https://localhost:8092 from my client laptop. But I can't access https://machine:8092 from my client laptop.

My network isn't set up for IPv6, so I can't check if I can access the DRP service on port 8092 using IPv6.

Is this expected? No connections from client machines to the server port 8092? Have I been completely sidetracked by the IPv6 thing?

**Comment by tombull** [comment]
I'm so sorry, I had the wrong IP address for the server machine, and a DNS issue on my local network... so I was connecting to the wrong machine - which was obviously refusing my connections on 8092.

I connected to the right machine on port 8092 and it connected fine.

Is there any way I can delete this embarrassing hour of wasting my time from the internet? Or ideally reset my brain and get the time I spent back?
